src/pcre2_jit_compile.c

```diff
@@ -9488,6 +9493,21 @@ if (common->utf && (*cc == OP_REFI || *cc == OP_DNREFI))
 
   CMPTO(SLJIT_EQUAL, TMP1, 0, char1_reg, 0, loop);
 
+  if (refi_flag & REFI_FLAG_TURKISH_CASING)
+    {
+    OP2(SLJIT_OR, SLJIT_TMP_DEST_REG, 0, char1_reg, 0, SLJIT_IMM, 0x20);
+    turkish_ascii_i = CMP(SLJIT_EQUAL, SLJIT_TMP_DEST_REG, 0, SLJIT_IMM, 0x69);
+
+    if (refi_flag & REFI_FLAG_CASELESS_RESTRICT)
+      add_jump(compiler, &no_match, turkish_ascii_i);
+
+    OP2(SLJIT_OR, SLJIT_TMP_DEST_REG, 0, char1_reg, 0, SLJIT_IMM, 0x1);
+    turkish_non_ascii_i = CMP(SLJIT_EQUAL, SLJIT_TMP_DEST_REG, 0, SLJIT_IMM, 0x131);
```

Comment:
I just remember that I read somewhere that 0x130/0x131 has no othercase by default. If this is true, then we don't need to do anything when REFI_FLAG_CASELESS_RESTRICT is set, and the second part can be simplified to:

```
if (!(refi_flag & REFI_FLAG_CASELESS_RESTRICT))
  {
  OP2(SLJIT_OR, SLJIT_TMP_DEST_REG, 0, char1_reg, 0, SLJIT_IMM, 0x1);
  turkish_non_ascii_i = CMP(SLJIT_EQUAL, SLJIT_TMP_DEST_REG, 0, SLJIT_IMM, 0x131);
  }
```

Reply:
I've tried to debug your code Zoltan, but I can't work it out.

It's clearly not doing the same thing as the C in pcre2_match.c's `match_ref()`, but I don't understand why you changed the logic.

I've spent a couple of hours going through the sljit tutorial, and its header file. I'll keep trying...

Reply:
The basic logic I can follow, and the overall structure seems correct & sensible.

But there's something about the "jump" instructions that isn't working the way I expect, or something like that?

Reply:
This is the test that's failing:

```
pcre2/build-linux$ ./pcre2test -16
PCRE2 version 10.45-DEV 2024-06-09 (16-bit)
  re> /(.) \1/ir,utf,ucp,jit
data> \x{212a} k
 0: \x{212a} k
 1: \x{212a}
data>
```

It's nothing to do with Turkish! There's just something broken in the caseless-restrict logic in the JIT.

Reply:
Yes. See the last comment.